src/PrestaShopBundle/Form/Admin/Improve/International/Localization/ImportLocalizationPackType.php

```diff
@@ -0,0 +1,178 @@
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class ImportLocalizationPackType is responsible for building 'Import a localization pack' form
+ * in 'Improve > International > Localization'
+ */
+class ImportLocalizationPackType extends TranslatorAwareType
+{
+    /**
+     * @var LocalizationPackLoaderInterface
+     */
+    private $remoteLocalizationPackLoader;
+
+    /**
+     * @var LocalizationPackLoaderInterface
+     */
+    private $localLocalizationPackLoader;
+
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param LocalizationPackLoaderInterface $remoteLocalizationPackLoader
+     * @param LocalizationPackLoaderInterface $localLocalizationPackLoader
+     * @param ConfigurationInterface $configuration
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        LocalizationPackLoaderInterface $remoteLocalizationPackLoader,
+        LocalizationPackLoaderInterface $localLocalizationPackLoader,
+        ConfigurationInterface $configuration
+    ) {
+        parent::__construct($translator, $locales);
+
+        $this->remoteLocalizationPackLoader = $remoteLocalizationPackLoader;
+        $this->localLocalizationPackLoader = $localLocalizationPackLoader;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('iso_localization_pack', ChoiceType::class, [
+                'choices' =>  $this->getLocalizationPackChoices(),
+                'attr' => [
+                    'data-toggle' => 'select2',
+                    'data-minimumResultsForSearch' => '7',
+                ],
+            ])
+            ->add('content_to_import', ChoiceType::class, [
+                'expanded' => true,
+                'multiple' => true,
+                'choices' => $this->getContentToImportChoices(),
+                'data' => [
+                    LocalizationPackImportConfig::CONTENT_STATES,
+                    LocalizationPackImportConfig::CONTENT_TAXES,
+                    LocalizationPackImportConfig::CONTENT_CURRENCIES,
+                    LocalizationPackImportConfig::CONTENT_LANGUAGES,
+                    LocalizationPackImportConfig::CONTENT_UNITS,
+                ],
+            ])
+            ->add('download_pack_data', SwitchType::class, [
+                'data' => 1,
+            ])
+        ;
+    }
+
+    /**
+     * Get available localization packs as choices
+     *
+     * @return array
+     */
+    private function getLocalizationPackChoices()
```

Comment:
Extract this private function to a final classe (and interface) and inject it instead: you will simplify A LOT your form type).

Something like `LocalisationPackChoices` with `all()` public function